src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/caseupdate/services/CaseUpdaterTest.java

```diff
@@ -330,4 +120,77 @@ public void should_not_update_case_data_without_any_documents() {
         assertThat(exception).hasMessage("Missing scanned documents in exception record");
         verifyNoInteractions(addressExtractor);
     }
+
+    private void assertCaseData(SampleCase actualCaseData, SampleCase expectedCaseData) {
+        assertThat(actualCaseData)
+            .isEqualToComparingOnlyGivenFields(expectedCaseData,
+                "legacyId",
+                "firstName",
+                "lastName",
+                "dateOfBirth",
+                "bulkScanCaseReference"
+            );
+    }
+
+    private void assertScannedDocumentsFromExistingCase(
+        List<Item<ScannedDocument>> actualScannedDocuments,
+        List<Item<ScannedDocument>> caseScannedDocuments
+    ) {
+        // Scanned documents in the updated case should contain scanned documents
+        // from the existing case and scanned documents from the exception record.
+        // Because there is 1 scanned document in the existing case
+        // the scanned document at index 0 in the updated case should match
+        // this scanned document in the existing case
+        assertThat(actualScannedDocuments.get(0).value)
+            .isEqualToComparingOnlyGivenFields(caseScannedDocuments.get(0).value,
+                "type",
+                "scannedDate",
+                "deliveryDate",
+                "fileName",
+                "exceptionRecordReference",
+                "controlNumber"
```

Comment:
or just excluding "document"?

Reply:
It appears that field-by-field comparison works here.